File: TEORIA/Ruby/FantasmaArcade/fantasma_arcade.rb
class FantasmaArcade

  @@velocidad = 0

  def initialize(color, posicion_x, posicion_y)
    @color = color
    @posicion_x = posicion_x
    @posicion_y = posicion_y
  end

  def self.setvelocidad(v)
    @@velocidad = v
  end

  def to_s
    "Color: #{@color}, Posición: (#{@posicion_x}, #{@posicion_y}), Velocidad: #{@@velocidad}"
  end
end

f = FantasmaArcade.new("rojo", 2, 3)
puts f.to_s()
FantasmaArcade.setvelocidad(5)
puts f.to_s()

File: TEORIA/Java/fantasmaarcade/FantasmaArcade.java
package fantasmaarcade;

public class FantasmaArcade {
    private static int velocidad;
    
    private String color;
    private int posicionX, posicionY;

    public FantasmaArcade(String color) {
        this.color = color;
        this.posicionX = 0;
        this.posicionY = 0;
    }
    
    public FantasmaArcade(String color, int posicionX, int posicionY) {
        this.color = color;
        this.posicionX = posicionX;
        this.posicionY = posicionY;
    }
    
    @Override
    public String toString(){
        return "Fantasma: " + color + " en posicion (" + posicionX + "," + posicionY + ") y velocidad " + velocidad; 
    }
    
    public static void setVelocidad(int vel){
        velocidad = vel;
    }
    
    public static void main(String[] args){
        FantasmaArcade f1 = new FantasmaArcade("Rojo", 2, 4);
        FantasmaArcade f2 = new FantasmaArcade("Blanco");
        System.out.println(f1.toString());
        System.out.println(f2.toString());
    }
}
